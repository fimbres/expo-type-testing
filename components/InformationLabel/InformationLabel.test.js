import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import { InformationLabel } from '.';

describe('<InformationLabel />', () => {
    it('render both texts correctly', () => {
        render(
            <InformationLabel
                primaryText='This is a test'
                secondaryText='And should pass'
                primaryTextStyle={{}}
                secondaryTextStyle={{}}
            />
        );

        const primaryText = screen.getByText('This is a test');
        const secondaryText = screen.getByText('And should pass');

        expect(primaryText).toBeDefined();
        expect(secondaryText).toBeDefined();
    });

    it('render correctly', () => {
        const tree = renderer
            .create(
                <InformationLabel
                primaryText='This is a test'
                secondaryText='And should pass'
                primaryTextStyle={{}}
                secondaryTextStyle={{}}
            />
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
