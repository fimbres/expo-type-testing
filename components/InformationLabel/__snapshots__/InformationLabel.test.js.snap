// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationLabel /> render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#9B9898",
          "fontFamily": "avenir-black",
          "fontSize": 14,
          "fontWeight": "800",
          "lineHeight": 19.12,
        },
        Object {},
      ]
    }
  >
    This is a test
  </Text>,
  <Text
    style={Object {}}
  >
    And should pass
  </Text>,
]
`;
